src/runtime_src/core/tools/common/SubCmdConfigureInternal.cpp

```diff
@@ -0,0 +1,462 @@
+      ("host", boost::program_options::value<decltype(m_host)>(&m_host), "IP or hostname for device peer")
+      ("security", boost::program_options::value<decltype(m_security)>(&m_security), "Update the security level for the device")
+      ("clk_throttle", boost::program_options::value<decltype(m_clk_throttle)>(&m_clk_throttle), "Enable/disable the device clock throttling")
+      ("ct_threshold_power_override", boost::program_options::value<decltype(m_power_override)>(&m_power_override), "Update the power threshold in watts")
+      ("ct_threshold_temp_override", boost::program_options::value<decltype(m_temp_override)>(&m_temp_override), "Update the temperature threshold in celsius")
+      ("ct_reset", boost::program_options::value<decltype(m_ct_reset)>(&m_ct_reset), "Reset all throttling options")
+      ("showx", boost::program_options::bool_switch(&m_showx), "Display the device configuration settings")
+    ;
+
+    m_hiddenOptions.add(configHiddenOptions);
+  }
+
+  m_commonOptions.add_options()
+    ("help", boost::program_options::bool_switch(&m_help), "Help to use this sub-command")
+  ;
+
+  auto configureDeviceTrees = JSONConfigurable::parse_configuration_tree({getConfigName()}, configurations);
+  auto currentConfigureTrees = JSONConfigurable::extractMatchingConfigurations<OptionOptions>(SubCmdConfigureInternal::optionOptionsCollection, configureDeviceTrees);
+  for (auto it = currentConfigureTrees.begin(); it != currentConfigureTrees.end(); it++) {
+    for (const auto& optionOptionsPtr : it->second)
+      addSubOption(optionOptionsPtr);
+  }
+}
+
+static config
+get_daemon_conf()
+{
+  config cfg;
+  cfg.host = xrt_core::utils::get_hostname();
+
+  std::ifstream istr(config_file);
+  if (!istr)
+    return cfg;
+
+  // Load persistent value, may overwrite default one
+  std::string line;
+  while (std::getline(istr, line)) {
+    auto pos = line.find('=', 0);
+    if (pos == std::string::npos)
+      throw xrt_core::system_error(EIO, "Bad daemon config file line '" + line + "'");
+    auto key = line.substr(0, pos++);
+    auto value = line.substr(pos);
+
+    if (key == "host") 
+      cfg.host = value;
+  }
+
+  return cfg;
+}
+
+/*
+ * helper function for option: showx
+ * shows daemon config
+ */
+static void 
+show_daemon_conf()
+{
+  auto cfg = get_daemon_conf();
+  std::cout << "Daemon:\n";
+  std::cout << boost::str(boost::format("  %s\n") % cfg);
+}
+
+/*
+ * helper function for option: showx
+ * shows device config
+ */
+static void 
+show_device_conf(xrt_core::device* device)
+{
+  auto bdf_raw = xrt_core::device_query<xrt_core::query::pcie_bdf>(device);
+  auto bdf = xrt_core::query::pcie_bdf::to_string(bdf_raw);
+
+  std::cout << bdf << "\n";
+
+  bool is_mfg = false;
+  bool is_recovery = false;
+  try {
+    is_mfg = xrt_core::device_query<xrt_core::query::is_mfg>(device);
+    is_recovery = xrt_core::device_query<xrt_core::query::is_recovery>(device);
```

Comment:
```suggestion
    is_recovery = xrt_core::device_query_default<xrt_core::query::is_recovery>(device, false);
```
Then move it out of the try/catch. Same for is_mfg